src/libcore/array.rs

```diff
@@ -41,21 +41,21 @@ pub trait FixedSizeArray<T> {
     fn as_mut_slice(&mut self) -> &mut [T];
 }
 
+impl<T, A: Unsize<[T]>> FixedSizeArray<T> for A {
+    #[inline]
+    fn as_slice(&self) -> &[T] {
+        unsafe { &*(self as *const A as *const [T]) }
```

Comment:
This can be just `self as &[T]` or in fact just `self` I think, and let coercion do the rest?